### PR TITLE
removed case sensitivity, fixed data overwrite, added missing comma

### PR DESCRIPTION
**Pull request by ds0440**
I will comment the git diff as well but during testing I found that case sensitivity for IAM actions was an issue. Amazon documentation says that the values of IAM actions are case insensitive [https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_action.html](url). When I used the `get_iam_actions` method I was not getting anything back even though my policy had actions that were in the IAM action list. This was caused by a case mismatch. The other changes were pretty minor and as mentioned I will note them in the diff.

### CODE REVIEW

File: pycfmodel/model/resources/properties/policy_document.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ class PolicyDocument(object):
         "IAM:CreateInstanceProfile",
         "IAM:PutUserPolicy",
         "IAM:ChangePassword",
-        "IAM:GenerateServiceLastAccessedDetails"
+        "IAM:GenerateServiceLastAccessedDetails",
```
Comment (ds0440) [inline pycfmodel/model/resources/properties/policy_document.py]: Added missing comma

File: pycfmodel/model/resources/properties/policy_document.py [hunk 2]
```diff
@@ -241,14 +241,14 @@ def get_iam_actions(self, difference=False):
...
                     actions = [
                         iam for iam in self.IAM_ACTIONS
-                        if iam.startswith(action[:wildcard])
+                        if action[:wildcard].lower() in iam.lower()
                     ]
                 elif wildcard == 0:
                     actions = self.IAM_ACTIONS
-                elif action in self.IAM_ACTIONS:
-                    actions = [action]
+                elif action.lower() in [iam.lower() for iam in self.IAM_ACTIONS]:
```
Comment (ds0440) [inline pycfmodel/model/resources/properties/policy_document.py]: This section addresses actions being overwritten on each iteration

**Comment by ds0440** [comment]
I meant to push this code under a different branch. Let me know if I need to separate these. I decided to move the list casting up into the class initialization. I did this because I found other places outside of statements that were expecting `action` to be a list and not a string. I also decided to do the same thing with the resource attribute. It looks like you were already handling `resource` not being  a list but I moved the type setting into the class initialization because I felt it helped to maintain the code conventions. I also added an empty list default for `action` since other parts of the code will expect it to be an a filled or empty list. Looking at this now I probably should have done the same with `resource`. I will update this now if need be or I will wait until my next commit.

File: pycfmodel/model/resources/properties/policy_document.py (hunk 2 above)
Comment (vlttnv) [reply pycfmodel/model/resources/properties/policy_document.py]: Ohh nice catch!

**Comment by vlttnv** [comment]
Awesome work! Looks good. The list casts make sense. We will have to make a change to our internal version of CFripper as one of the rules expects `resource` to be a string, but that's on our side.

Again, may I ask you to bump the version. It might make sense to bump it to `0.3.0` as there is a breaking change present.
Apologies for the need for manual bumps but we don't have an auto version bumper in place.
Thanks for your contributions!

**Comment by ds0440** [comment]
Thanks! I honestly wasn't sure how all the changes would be received. I will bump the version and add the `resource` default of an empty list.

**Comment by ds0440** [comment]
This weird. It doesn't seem like the test integration is coming out of the pending state. If you look at the details of the Travis CI build it says everything passes. Not sure if we might need to run it again?

**Pull request merged**